lobster/tools/core/online_report.py

```diff
@@ -32,6 +32,98 @@ class Parse_Error(Exception):
     pass
 
 
+def is_git_main_module(path):
+    return os.path.isdir(os.path.join(path, ".git"))
+
+
+def is_dir_in_git_submodule(directory):
+    """
+    Checks if a given directory is nested inside a Git submodule.
+
+    Args:
+        directory (str): The path to the directory to check.
+
+    Returns:
+        bool: True if the directory is inside a Git submodule,
+        False otherwise.
+        str: The path to the superproject of submodule.
+    """
+    try:
+        # Check if the directory is part of a Git submodule
+        result = subprocess.run(['git',
+                                 'rev-parse',
+                                 '--show-superproject-working-tree'],
+                                cwd=directory,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                universal_newlines=True,
+                                check=True)
+        if result.returncode == 0 and result.stdout.strip():
+            return True, result.stdout.strip()
+        else:
+            return False, ''
+    except (subprocess.CalledProcessError, OSError):
+        return False, ''
+
+
+def is_dir_in_git_main_module(directory):
+    """
+    Checks if a given directory is nested inside a Git main module.
+
+    Args:
+        directory (str): The path to the directory to check.
+
+    Returns:
+        bool: True if the directory is inside a Git mainmodule,
+        False otherwise.
+        str: The path to the mainmodule.
+    """
+    try:
+        # Check if the directory is part of a Git main module
+        result = subprocess.run(['git', 'rev-parse', '--show-toplevel'],
+                                cwd=directory,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                universal_newlines=True,
+                                check=True)
+        if result.returncode == 0 and result.stdout.strip():
+            return True, result.stdout.strip()
+        else:
+            return False, ''
```

Comment:
You could also return `None` instead of `''`.
But actually one return value would be enough. Instead of 
```python
return True, result.stdout.strip()
```
just do
```python
return result.stdout.strip()
```
and instead of
```python
return False, ''
```
just do
```python
return None
```